### ISSUE TITLE
Air tiles

### ISSUE THREAD

**Issue by dwidge** [AlexeyBond/godot-constraint-solving#14]
Hello, how to make it use empty space in the sample? With walls in a GridMap it gave an error assert(mapper.size()>1) because it was only one tile type. Actually there are two types: wall and empty. Add another wall type, it runs but it fills target region with solid wall, ignores free space in positive sample.

**Comment by AlexeyBond** [comment]
Hi. That's an expected behavior. In most cases an empty space is not useful, so, if it wasn't ignored in sample map, it would be more difficult to create a valid sample for rules that don't use empty tiles. With empty space ignored it is easy to just throw some new tiles somewhere in free space of the map and add some adjacent tiles like this:
![image](https://github.com/AlexeyBond/godot-constraint-solving/assets/7383049/c80529dd-37ca-4e34-8042-17af6cf17e8b)
without spending time to fit them into existing part of the map. If empty space wasn't ignored, a sample for a map that doesn't use empty tiles would have always been kept a rectangle without any gaps, because otherwise those gaps would be recognized as  empty tiles.

It may be possible to add use of empty cells as an option. But I'm not yet sure how to make it work without interfering with functionality of negative samples, preconditions and communication between threads in multithreaded mode.

---

The cleanest solution I currently see is to create a tile type that is visible in editor but isn't visible in the game. And use this tile to represent an empty space in samples.

I have just researched how to make it work with grid maps. And the only way I found is a bit complicated and not very efficient (as the "empty" mesh is rendered in game):

- add a global boolean shader uniform `IS_EDITOR`, defaults to true
- in some game script it should be set to `false`: `RenderingServer.global_shader_parameter_set("IS_EDITOR", false)`
- create new material for "empty" mesh. It should use shader like
  ```glsl
  shader_type spatial;
  global uniform bool IS_EDITOR;
  
  void fragment() {
  	if (!IS_EDITOR) {
  		discard; // Do not render in actual game
  	}
  
  	ALBEDO = vec3(1.0,0.0,0.0); // Mesh will appear red in the editor
  }
  ```
- add a mesh with this new material to mesh library (use surface material override, not material override if you're using a mesh imported with different material)

Now this mesh can be used in samples where an empty space is supposed to be. And in the game the space will look empty.

This could be easier if meshlib/gridmap did respect visibility layers set in meshlib source scene, but they don't.
